corehq/project_limits/models.py

```diff
@@ -97,3 +97,15 @@ class Meta:
     def __str__(self):
         domain = f"[{self.domain}] " if self.domain else ""
         return f"{domain}{self.key}: {self.limit}"
+
+    @classmethod
+    def for_key(cls, key, domain=None):
+        """
+        Return the value associated with the given key, prioritizing specificity
+        """
+        domain_filter = models.Q(domain="")
+        if domain:
+            domain_filter |= models.Q(domain=domain)
+        filters = models.Q(key=key) & domain_filter
+        limit = SystemLimit.objects.filter(filters).order_by("-domain").values_list("limit", flat=True).first()
+        return limit or None
```

Comment:
What are the possible values of `limit` that evaluate to false?

Reply:
Ah great catch. It is a positive integer field, so 0 is the only possible falsey value in addition to None. Updated to fix that [8bbb387](https://github.com/dimagi/commcare-hq/pull/35781/commits/8bbb387a888396cd3858e34bb4ba5c040ff48a4a)